src/backend/access/aocs/aocsam_handler.c

```diff
@@ -2310,6 +2310,15 @@ aoco_transform_column_encoding_clauses(Relation rel, List *aocoColumnEncoding,
 	char	   *compresstype = NULL;
 	NameData	compresstype_nd;
 
+	/*
+	 * The relam of partition table may be ao table, but partition table
+	 * has no entry in pg_appendonly. It shouldn't fetch encoding options
+	 * from here for partition tables. See details in function
+	 * transformColumnEncoding.
+	 */
+	if (rel && rel->rd_rel->relkind == RELKIND_PARTITIONED_TABLE)
+		rel = NULL;
```

Comment:
use rel = NULL to control the logical is not the best way.

Reply:
Do you have any suggestions?
Whether rel is null has different code branch in the following code. It is ao-specific implementation. AO partition tables have no entry in pg_appendonly. We should avoid touch pg_appendonly for partition table.